File: jbake-core/src/main/java/org/jbake/parser/ErrorEngine.java
package org.jbake.parser;

import org.jbake.model.DocumentModel;

import java.time.Instant;

/**
 * An internal rendering engine used to notify the user that the markup format he used requires an engine that couldn't
 * be loaded.
 *
 * @author Cédric Champeau
 */
public class ErrorEngine extends MarkupEngine {
    private final String engineName;

    public ErrorEngine() {
        this("unknown");
    }

    public ErrorEngine(final String name) {
        engineName = name;
    }

    @Override
    public void processHeader(final ParserContext context) {
        DocumentModel documentModel = context.getDocumentModel();
        documentModel.setType("post");
        documentModel.setStatus("published");
        documentModel.setTitle("Rendering engine missing");
        documentModel.setDate(Instant.now());
        documentModel.setTags(new String[0]);
    }

    @Override
    public void processBody(final ParserContext context) {
        context.setBody("The markup engine [" + engineName + "] for [" + context.getFile() + "] couldn't be loaded");
    }
}

File: jbake-core/src/main/java/org/jbake/parser/ErrorEngine.kt
package org.jbake.parser

import java.util.*

/**
 * An internal rendering engine used to notify the user that the markup format he used requires an engine that couldn't
 * be loaded.
 */
class ErrorEngine @JvmOverloads constructor(private val engineName: String = "unknown")
    : MarkupEngine()
{
    override fun processHeader(context: ParserContext) {
        val documentModel = context.documentModel
        documentModel.type = "post"
        documentModel.status = "published"
        documentModel.title = "Rendering engine missing"
        documentModel.date = Date()
        documentModel.tags = emptyArray<String>()
    }

    override fun processBody(parserContext: ParserContext) {
        parserContext.body = "The markup engine [" + engineName + "] for [" + parserContext.file + "] couldn't be loaded"
    }
}
